### ISSUE TITLE
Completion generation from pandoc manpage is broken

### ISSUE THREAD

**Issue by daenney** [fish-shell/fish-shell#2937]
### Reproduction Steps:

1. Install pandoc and put the manpage on the `MANPATH`
2. `fish_update_completion`
3. `pandoc -<tab>`

### Expected behavior:

Get completions for the long and short options in the man page.

### Observed behavior:

Get completions for long and short options in the man page riddled with a lot of garbage:

```
-\).                                            (Here is a simple example: * one * two * three T… [See Man Page])
---                                                                                                    (author:.)
---\)                                                      (In reading LaTeX, parse the characters `, [aq], and.)
-----  -----:                                                                  (apple|2. 05 pear|1. 37 orange|3.)
------  -------  ----------                                                    (12 12 12 12 123 … [See Man Page])
-----------  -------------------------                                                            (First row 12.)
---------------                                          ((This syntax is compatible with PHP Markdown Extra. ).)
------------------                              (The header text can contain inline formatting, … [See Man Page])
-------------------------------------------------------------  (Centered Default Right Left Head… [See Man Page])
--ascii                                                                    (Use only ascii characters in output.)
```
---
Fish version: `fish, version 2.2.0`

Operating system: OS X 10.11 and Homebrew

Terminal or terminal emulator: iTerm2

**Comment by daenney** [comment]
I've gisted the man page here: https://gist.github.com/daenney/4cd81d2795cd5221b0a06e4277a8c78e

**Comment by faho** [comment]
Yes, it's borked. Seems we choke on some of the YAML-blocks.

Though my preferred solution would be just writing a pandoc completion - there's a few things here that the generated ones won't do anyway, like formats, and it doesn't seem too complicated.

**Comment by daenney** [comment]
I agree a more complete one should be written by hand. In this case though as long as we could skip the YAML blocks I think that would solve most problems.

**Comment by krader1961** [comment]
This is not a bug other than in the most narrow sense. Completions generated from man pages is always best effort and seldom optimal. Improving the parsing of random man pages is an enhancement. Although, frankly in this case I think it makes more sense for someone who uses `pandoc` to write a custom completion script.

**Comment by dsanson** [comment]
I've written a completion for pandoc. I'm new to fish, so doubtless there are ways it could be improved.

https://github.com/dsanson/fish-completion-pandoc

**Comment by zanchey** [comment]
@dsanson, would you be willing to submit that for inclusion in fish? It would mean relicensing as GPL 2.

**Comment by dsanson** [comment]
@zanchey: I changed the license to GPL 2. What is the process ofr submitting it for inclusion?

**Comment by faho** [comment]
@dsanson: A PR that adds it to [share/completions/](https://github.com/fish-shell/fish-shell/tree/master/share/completions) would work. Or do you mean something else?
